pkg/provider/provider.go

```diff
@@ -458,20 +473,32 @@ func (k *kubeProvider) Update(
 	newInputs := propMapToUnstructured(newResInputs)
 
 	// Apply update.
-	liveObj, err := await.Update(k.pool, k.client, oldInputs, newInputs)
-	if err != nil {
-		return nil, err
+	initialized, awaitErr := await.Update(k.pool, k.client, oldInputs, newInputs)
+	if awaitErr != nil {
+		var getErr error
+		initialized, getErr = k.readLiveObject(newInputs)
```

Comment:
I think maybe I'm not understanding the semantics of `readLiveObject`. I see that `readLiveObject` keys off the resource name in order to read it, but is it always the case here that the resource name in `newInputs` (uniquely) refers to the resource that we just failed to create?

I'm thinking this is OK, I'm just trying to wrap my head around this.

Reply:
I guess what's on my mind is that it seems to me that an object's fully qualified name is the ID that is ultimately sent to the engine, but here's we're just using a regular name?

Reply:
@swgillespie Yes, it should always uniquely refer to the resource we failed to create. This will change when we address #10.